sklearn/feature_selection/_base.py

```diff
@@ -121,3 +124,47 @@ def inverse_transform(self, X):
         Xt = np.zeros((X.shape[0], support.size), dtype=X.dtype)
         Xt[:, support] = X
         return Xt
+
+
+def _get_feature_importances(estimator, getter, transform_func=None,
+                             norm_order=1):
+    """Retrieve or aggregate feature importances from estimator"""
+    if isinstance(getter, str):
+        if getter == 'auto':
+            if hasattr(estimator, 'coef_'):
+                getter = attrgetter('coef_')
+            elif hasattr(estimator, 'feature_importances_'):
+                getter = attrgetter('feature_importances_')
+            else:
+                raise ValueError(
+                    f"when `importance_getter=='auto'`, the underlying "
+                    f"estimator {estimator.__class__.__name__} should have "
+                    f"`coef_` or `feature_importances_` attribute. Either "
+                    f"pass a fitted estimator to feature selector or call fit "
+                    f"before calling transform."
+                )
+        else:
+            getter = attrgetter(getter)
+    elif not callable(getter):
+        raise ValueError(
+            '`importance_getter` has to be a string or `callable`'
+        )
+    importances = getter(estimator)
+
+    if transform_func == "norm":
+        if importances.ndim == 1:
+            importances = np.abs(importances)
+        else:
+            importances = np.linalg.norm(importances, axis=0,
+                                         ord=norm_order)
+    elif transform_func == "square":
+        if importances.ndim == 1:
+            importances = safe_sqr(importances)
+        else:
+            importances = safe_sqr(importances).sum(axis=0)
+    elif not transform_func:
```

Comment:
Oh I see. I prefer `transform_func is None` and an `else`. This is more explicit.

Reply:
ya, Make sense.